src/SecTester.Scan/CI/VendorMatcher.cs

```diff
@@ -0,0 +1,88 @@
+using System.Collections;
+using System.Linq;
+using System.Text.Json;
+
+namespace SecTester.Scan.CI;
+
+internal class VendorMatcher
+{
+  private readonly IDictionary _env;
+
+  public VendorMatcher(IDictionary env)
+  {
+    _env = env;
+  }
+
+  public bool MatchEnvElement(JsonElement element)
+  {
+    return element.ValueKind switch
+    {
+      JsonValueKind.String => element.GetString() is not null && _env.Contains(element.GetString()!),
```

Comment:
It may lead to FP in the case of an empty string. Also, you have to pass `StringComparison` second argument